UICatalog/Scenarios/SplitContainerExample.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Terminal.Gui;
+using System;
+using Terminal.Gui.Graphs;
+using NStack;
+using System.Linq;
+
+namespace UICatalog.Scenarios {
+	[ScenarioMetadata (Name: "Split Container", Description: "Demonstrates the SplitContainer functionality")]
+	[ScenarioCategory ("Controls")]
+	[ScenarioCategory ("LineView")]
+	public class SplitContainerExample : Scenario {
+
+		private SplitContainer splitContainer;
+
+		private SplitContainer splitContainer2;
+		private MenuItem miVertical;
+		private MenuItem miShowBoth;
+		private MenuItem miShowPanel1;
+		private MenuItem miShowPanel2;
+		private MenuItem miShowNeither;
+
+		/// <summary>
+		/// Setup the scenario.
+		/// </summary>
+		public override void Setup ()
+		{
+			// Scenario Windows.
+			Win.Title = this.GetName ();
+			Win.Y = 1;
+
+			Win.Add (new Label ("This is a SplitContainer with a minimum panel size of 4. Drag the splitter to resize:"));
+
+			splitContainer = new SplitContainer {
+				Y = 2,
+				X = 2,
+				Width = Dim.Fill () - 2,
+				Height = Dim.Fill () - 1,
+				SplitterDistance = Pos.Percent (50), // TODO: get this to work with drag resizing and percents
+			};
+			splitContainer2 = new SplitContainer(){
+				Width = Dim.Fill(),
+				Height = Dim.Fill(),
+				Orientation = Orientation.Horizontal
+			};
+
+			splitContainer.Panels [0].MinSize = 4;
+			splitContainer.Panels [1].MinSize = 4;
+
+			Label lbl1;
+			splitContainer.Panels [0].Title = "Hello";
+			splitContainer.Panels [0].Add (lbl1 = new Label ("Type Something:") { Y = 0 });
+			splitContainer.Panels [0].Add (new TextField () { Width = Dim.Fill (), Y = 0, X = Pos.Right (lbl1) + 1 });
+
+			Label lbl2;
+			splitContainer.Panels [1].Title = "World";
+			splitContainer.Panels[1].Add(splitContainer2);
+
+			splitContainer2.Panels [0].Add (new TextView ()
```

Comment:
The line-ends on the embedded split container are not drawing as Ts correctly. They really should!

Reply:
With the current way this is architected it would be difficult for the nested container to draw these line endings.

While it looks like there is a single container with 2 dividing lines there is in fact 2 seperate containers nested inside each other.  The nested container is configured to have no border so it looks like it integrates with the parent.  To draw a 'T' into the parent it would have to draw outside its bounds

![image](https://user-images.githubusercontent.com/31306100/210910811-59b35419-dc46-4109-80d5-8178f5a4aef3.png)
_Pink rectangle highlights the bounds of the parent container.  Green rectangle highlights the bounds of the nested container_

Things would get more complicated when considering a 2x2 configuration.  We would need a 'crosshair' in the middle but only until one of the two horizontal lines were moved.

![image](https://user-images.githubusercontent.com/31306100/210911031-de1233c5-5110-4aec-931e-dd4b7cf42995.png)
_This would require a crosshair for the intersecting line_

After the horizontal line (e.g. left) is moved up we would need two Ts again.

![image](https://user-images.githubusercontent.com/31306100/210911164-3df66ec9-a000-4f32-9c14-b8c670acd6f6.png)
_After moving the nested container up we would need two T instead of a crosshair_

Reply:
I went down the "DrawWindowFrame With Panes" hole because of this. 

Another idea is to make `SplitContainer` support arbitrary panes without them being nested.

Reply:
Fully implemented now, sub split containers all join up nicely now.